core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/DiagnosticLog.java

```diff
@@ -102,4 +127,246 @@ public Map<String, Object> getConfigurations() {
 
         return configurations;
     }
+
+    /**
+     * Get the log level of the diagnostic log.
+     *
+     * @return log level
+     */
+    public LogLevel getLogLevel() {
+
+        return logLevel;
+    }
+
+    /**
+     * Log levels of the diagnostic log.
+     */
+    public enum LogLevel {
+
+        BASIC, // This level is intended for App Developers (and Internal Developers).
+        ADVANCED // This level is intended only for Internal Developers.
+    }
+
+    /**
+     * Result status of the diagnostic log.
+     */
+    public enum ResultStatus {
+
+        SUCCESS, // Successful execution.
+        FAILED, // Failed execution.
+        PENDING, // Pending execution.
+        NONE // No result status.
```

Comment:
do we need these statuses? I feel like this could lead to inconsistencies with the usages. Currently, we add "SUCCESS" status to all the non-error flows. If we introduce something like PENDING, NONE, we would need to change the already existing flows as well. But, I don't really see a need to include these. WDYT?

cc: @pulasthi7

Reply:
removed unwanted status with 6d2d59fb69b1ebac978aa76cacc1d441b6c0d204